lib/polymorphic_integer_type/belongs_to_polymorphic_association_extension.rb

```diff
@@ -3,7 +3,7 @@ module Associations
     class BelongsToPolymorphicAssociation < BelongsToAssociation
       private def replace_keys(record)
         super
-        owner[reflection.foreign_type] = record.class.base_class
+        owner[reflection.foreign_type] = record.class
```

Comment:
send the class object instead of the base_class so we can try several methods like `#polymorphic_name`, `#sti_name`, `#to_s` etc